packages/components/src/ListBox/src/ListBoxItem.tsx

```diff
@@ -0,0 +1,313 @@
+import { CheckmarkIcon, IconContext, type IconProps } from "@hopper-ui/icons";
+import { type StyledComponentProps, useStyledSystem, type ResponsiveProp, useResponsiveValue } from "@hopper-ui/styled-system";
+import { forwardRef, type ReactNode, useContext, type ForwardedRef, useState, useEffect, useCallback } from "react";
+import {
+    useContextProps, 
+    ListBoxItem as RACListBoxItem, 
+    type ListBoxItemProps as RACListBoxItemProps, 
+    composeRenderProps,
+    DEFAULT_SLOT,
+    ListStateContext,
+    type ListBoxItemRenderProps
+} from "react-aria-components";
+
+import { BadgeContext } from "../../Badge/index.ts";
+import { Checkbox } from "../../checkbox/index.ts";
+import { IconListContext } from "../../IconList/index.ts";
+import { Radio, RadioGroup, RadioList } from "../../radio/index.ts";
+import { Text, TextContext, type TextProps } from "../../typography/Text/index.ts";
+import { composeClassnameRenderProps, type SizeAdapter, SlotProvider, cssModule, isTextOnlyChildren } from "../../utils/index.ts";
+
+import { ListBoxItemContext } from "./ListBoxItemContext.ts";
+
+import styles from "./ListBoxItem.module.css";
+
+export const GlobalListBoxItemCssSelector = "hop-ListBoxItem";
+
+export type ListBoxItemSize = "xs" | "sm" | "md" | "lg";
+
+export interface ListBoxItemProps<T> extends StyledComponentProps<Omit<RACListBoxItemProps<T>, "orientation | layout">> {
+    /**
+     * Whether or not the ListBoxItem is in an invalid state.
+     */
+    isInvalid?: boolean;
+    /**
+     * The selection indicator to use. Only available if the selection mode is not "none".
+     * When set to "input", the selection indicator will be an either a radio or checkbox based on the selection mode.
+     * @default "check"
+     */
+    selectionIndicator?: "check" | "input";
+    /**
+     * A ListBoxItem can vary in size.
+     * @default "sm"
+     */
+    size?: ResponsiveProp<ListBoxItemSize>;
+}
+
+interface ListBoxInnerProps extends ListBoxItemRenderProps {
+    /**
+     * The selection indicator to use. Only available if the selection mode is not "none".
+     * When set to "input", the selection indicator will be an either a radio or checkbox based on the selection mode.
+     * @default "check"
+     */
+    selectionIndicator?: "check" | "input";
+    /**
+     * Whether or not the ListBoxItem is in an invalid state.
+     */
+    isInvalid?: boolean;
+    /**
+     * A ListBoxItem can vary in size.
+     * @default "sm"
+     */
+    size: ListBoxItemSize;
+    /**
+     * The children of the ListBoxItem.
+     */
+    children: ReactNode;
+    /**
+     * A function to set the list has selection state.
+     */
+    setListHasSelection: (value: boolean) => void;
```

Comment:
but we added this so that we don't need to use the listbox class in the list box item css. For the checkmark animation, we need to know when there's at least one selection.

Reply:
ya i know, i put a comment under with a suggestion